src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -160,32 +192,64 @@ [.. surroundingStatements
             {
                 var usingStatement = CreateUsingStatement(declarationStatement, statementsToSurround);
 
-                if (statementsToSurround.Any())
-                {
-                    var newParent = WithStatements(
-                        declarationStatement.GetRequiredParent(),
-                        [.. surroundingStatements
-                            .Take(declarationStatementIndex)
-                            .Concat(usingStatement)
-                            .Concat(surroundingStatements.Skip(declarationStatementIndex + 1 + statementsToSurround.Count))]);
-
-                    return document.WithSyntaxRoot(root.ReplaceNode(
-                        declarationStatement.GetRequiredParent(),
-                        newParent.WithAdditionalAnnotations(Formatter.Annotation)));
-                }
-                else
-                {
-                    // Either the parent is not blocklike, meaning WithStatements can’t be used as in the other branch,
-                    // or there’s just no need to replace more than the statement itself because no following statements
-                    // will be surrounded.
-                    return document.WithSyntaxRoot(root.ReplaceNode(
-                        declarationStatement,
-                        usingStatement.WithAdditionalAnnotations(Formatter.Annotation)));
-                }
+                return await ReplaceWithUsingStatementAsync(
+                    document, declarationStatement, statementsToSurround, usingStatement, cancellationToken).ConfigureAwait(false);
             }
         }
     }
 
+    private async Task<Document> IntroduceUsingStatementAsync(
+        Document document,
+        TExpressionStatementSyntax expressionStatement,
+        CancellationToken cancellationToken)
+    {
+        var surroundingStatements = GetSurroundingStatements(expressionStatement);
+        var statementIndex = surroundingStatements.IndexOf(expressionStatement);
+
+        SyntaxList<TStatementSyntax> statementsToSurround = [.. surroundingStatements.Skip(statementIndex + 1)];
+
+        var usingStatement = CreateUsingStatement(expressionStatement, statementsToSurround);
```

Comment:
>var usingStatement = CreateUsingStatement(expressionStatement, statementsToSurround); [](http://example.com/codeflow?start=8&length=85)

nit: consider moving inside ReplaceWithUsingStatementAsync

Reply:
Different types of using statements are created (i updated their names to make this clearer).